libs/coin-modules/coin-evm/src/__tests__/coin-tester/scenarios/ethereum.ts

```diff
@@ -0,0 +1,200 @@
+import Eth from "@ledgerhq/hw-app-eth";
+import { BigNumber } from "bignumber.js";
+import { ethers, providers } from "ethers";
+import { killSpeculos, spawnSigner } from "@ledgerhq/coin-tester/docker";
+import { encodeTokenAccountId } from "@ledgerhq/coin-framework/account/index";
+import { Scenario, ScenarioTransaction } from "@ledgerhq/coin-tester/main";
+import { getCryptoCurrencyById } from "@ledgerhq/cryptoassets/currencies";
+import { buildAccountBridge, buildCurrencyBridge } from "../../../bridge/js";
+import { makeAccount } from "../../fixtures/common.fixtures";
+import { EvmNftTransaction, Transaction as EvmTransaction } from "../../../types";
+import resolver from "../../../hw-getAddress";
+import { setCoinConfig } from "../../../config";
+import {
+  ethereum,
+  ERC20Interface,
+  USDC_ON_ETHEREUM,
+  ERC721Interface,
+  ERC1155Interface,
+} from "../helpers";
+import { clearExplorerAppendix, getLogs, setBlock } from "../indexer";
+import { killAnvil, spawnAnvil } from "../docker";
+
+const scenarioSendTransaction: ScenarioTransaction<Partial<EvmTransaction>> = {
+  name: "Send ethereum",
+  amount: new BigNumber(100),
+  recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+};
+
+// use function createTransaction
+const scenarioUSDCTransaction: ScenarioTransaction<Partial<EvmTransaction>> = {
+  name: "Send USDC",
+  amount: new BigNumber(100),
+  recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+  subAccountId: encodeTokenAccountId(
+    "js:2:ethereum:0x3313797c7B45F34c56Bdedc0179992A4d435AF25:",
+    USDC_ON_ETHEREUM,
+  ),
+};
+
+const scenarioERC721Transaction: ScenarioTransaction<Partial<EvmTransaction & EvmNftTransaction>> =
+  {
+    name: "Send NFT",
+    recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+    mode: "erc721",
+    nft: {
+      tokenId: "3368",
+      contract: "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
+      quantity: new BigNumber(1),
+      collectionName: "Bored Ape",
+    },
+  };
+
+const scenarioERC1155Transaction: ScenarioTransaction<Partial<EvmTransaction & EvmNftTransaction>> =
+  {
+    name: "Send ERC1155",
+    recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+    mode: "erc1155",
+    nft: {
+      tokenId: "951",
+      collectionName: "Clone X",
+      // EIP55 checksum of 0x348fc118bcc65a92dc033a951af153d14d945312
+      contract: "0x348FC118bcC65a92dC033A951aF153d14D945312",
+      quantity: new BigNumber(2),
+    },
+  };
+
+const defaultNanoApp = { firmware: "2.2.3" as const, version: "1.10.4" as const };
+
+export const scenarioEthereum: Scenario<EvmTransaction> = {
```

Comment:
Scenario should be implementing a `beforeAll` and `afterAll` acknowledging the account state ("account has X balance, Y tokens, Z NFTs"...) to confirm the correct execution of the smart contracts